### PR TITLE
feat: update build-canned

### PR DESCRIPTION
**Pull request by dovholuknf**
updates compose-builder's `make build-canned`

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

File: compose-builder/Makefile [hunk 1]
```diff
@@ -1035,7 +1038,7 @@ build-taf-nanomq:
 	make taf-compose-perf taf-perf-no-secty mqtt-bus nanomq no-secty
 
 compose: gen
-	cat gen-header docker-compose.yml > $(RELEASE_FOLDER)/docker-compose$(NO_SECURITY)$(APP_SAMPLE)$(BUS)$(NANOMQ)$(ARCH).yml
+	cat gen-header docker-compose.yml > $(RELEASE_FOLDER)/docker-compose$(NO_SECURITY)$(APP_SAMPLE)$(BUS)$(NANOMQ)$(ARCH)$(ZERO_TRUST_COMPOSE).yml
```
Comment (lenny-intel) [inline compose-builder/Makefile:1041]: ARCH needs to always be last. 

```suggestion
	cat gen-header docker-compose.yml > $(RELEASE_FOLDER)/docker-compose$(NO_SECURITY)$(APP_SAMPLE)$(BUS)$(NANOMQ)$(ZERO_TRUST_COMPOSE)$(ARCH).yml
```

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Review by lenny-goodell — DISMISSED** [review]
LGTM, putting in draft until @cloudxxx8 and team can agree to the required min version of docker compose.

**Comment by dovholuknf** [comment]
@cloudxxx8 - I've updated compose builder with the alpha release of ekuiper that contains OpenZiti support. When the actual build is released, we can update again. I've asked about the release date of ekuiper 1.14 here https://github.com/lf-edge/ekuiper/discussions/2772

In a previous core WG meeting we agreed verbally and in the TSC we agreed that the version of docker compose was acceptable. I'm marking this PR as 'ready'

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Comment by cloudxxx8** [comment]
recheck

**Comment by cherrycl** [comment]
@dovholuknf Would you please rebase and squash this PR to avoid unknown error for smoke test? Thanks.

**Comment by dovholuknf** [comment]
@cherrycl i have rebased and squashed

**Comment by cherrycl** [comment]
@dovholuknf I saw your branch still displayed `1 commit behind edgexfoundry/edgex-compose:main`, would you please rebase main branch again?

**Comment by dovholuknf** [comment]
@cherrycl -- all set. `This branch is 1 commit ahead of edgexfoundry/edgex-compose:main.`

**Pull request merged**